README.md

```diff
@@ -2,22 +2,84 @@
 
 Functionality for fetching index provider peer IDs for a minder id on the Filecoin network.
 
-## Constants
+## Usage
 
-### Smart Contract Integration
+### Getting a Miner's PeerId
+
+The library provides a function to retrieve a miner's PeerId by querying both the Filecoin smart
+contract and FilecoinMinerInfo. It will automatically try both sources and return the first valid
+result.
+
+```js
+    maxAttempts = 5,
+import { getIndexProviderPeerId, MINER_TO_PEERID_CONTRACT_ADDRESS, MINER_TO_PEERID_CONTRACT_ABI } from '@filecoin-station/index-provider-peer-id';
+import { ethers } from 'ethers';
+
+// Initialize your ethers contract instance
+const provider = new ethers.providers.JsonRpcProvider('https://eth.example.com');
+const contractAddress = MINER_TO_PEERID_CONTRACT_ADDRESS;
+const contractAbi = MINER_TO_PEERID_CONTRACT_ABI;
+const contract = new ethers.Contract(contractAddress, contractAbi, provider);
+
+// Basic usage
+const minerId = 'f0142637';
+const peerId = await getIndexProviderPeerId(minerId, contract);
+console.log(peerId); // e.g., '12D3KooWMsPmAA65yHAHgbxgh7CPkEctJHZMeM3rAvoW8CZKxtpG'
+
+// Advanced usage with options
+const peerIdWithOptions = await getIndexProviderPeerId(minerId, contract, {
+  maxAttempts: 3,                             // Number of retry attempts (default: 5)
+  rpcUrl: 'https://custom-filecoin-api.com',  // Custom Filecoin RPC endpoint (default: 'https://api.node.glif.io/')
+  rpcAuth: 'your-auth-token',                 // Optional authorization token for RPC
+  rpcFn: customRpcFunction                    // Optional custom RPC function implementation.
+});
+```
+
+The function prioritizes the smart contract result. If the smart contract doesn't return a valid
+PeerId, it falls back to checking FilecoinMinerInfo. If both methods fail, an error is thrown with
```

Comment:
```suggestion
PeerId, it falls back to the result from FilecoinMinerInfo. If both methods fail, an error is thrown with
```

The current documentation isn't exactly correct, as the lib tries both concurrently, and then decides what to return.